internal/daemon/controller/handler.go

```diff
@@ -57,18 +58,113 @@ type HandlerProperties struct {
 	CancelCtx      context.Context
 }
 
+type ResponseWriter struct {
+	// TODO: this is the easiest way for me to do this, but there are a couple issues with this.
+	// If boundary does not call WriteHeader, but only Write, then http.ResponseWriter will
+	// internally called WriterHeader, and ours will be skipped, and the headers will not be
+	// written. This is also an issue for requests sent with an "Expect: 100-continue" header,
+	// Where WriteHeader is called automatically
+	wrapped http.ResponseWriter
+	// headers contain a map of response code to header map such that
+	// headers[status][header name] = header value
+	// this map also contains values for hundred-level values in the format "1xx", "2xx", etc
+	headers map[string]map[string]string
+}
+
+func (w *ResponseWriter) WriteHeader(statusCode int) {
+	w.setCustomResponseHeaders(statusCode)
+	w.wrapped.WriteHeader(statusCode)
+}
+
+func (w *ResponseWriter) Header() http.Header {
+	return w.wrapped.Header()
+}
+
+func (w *ResponseWriter) Write(data []byte) (int, error) {
+	// TODO: if WriteHeader has not been called yet, this will bypass it and called the internal
+	// WriteHeader func, without adding our headers.
+	return w.wrapped.Write(data)
+}
+
+func (w *ResponseWriter) setCustomResponseHeaders(statusCode int) {
+	sch := w.headers
+	if sch == nil {
+		return
+	}
+
+	// Check the validity of the status code
+	if statusCode >= 600 || statusCode < 100 {
+		return
+	}
+
+	// Setter function to set headers
+	setter := func(headerMap map[string]string) {
+		for header, value := range headerMap {
+			w.Header().Set(header, value)
+		}
+	}
+
+	// Setting the default headers first
+	if val, ok := sch["default"]; ok {
+		setter(val)
+	}
+
+	// Then setting the generic hundred-level headers
+	d := fmt.Sprintf("%vxx", statusCode/100)
```

Comment:
Use `%d` here to ensure we're working with integers. `%v` allows any input value.
```suggestion
	d := fmt.Sprintf("%dxx", statusCode/100)
```